iroh-s3-bao-store/src/lib.rs

```diff
@@ -41,11 +48,18 @@ impl S3Store {
         let mut http_adapter = HttpAdapter::new(url.clone());
         let data = http_adapter.read_to_end().await?;
         let size = data.len() as u64;
-        let (mut outboard, hash) = bao_tree::io::outboard(data, IROH_BLOCK_SIZE);
+        let (mut outboard, hash) = {
+            let outboard = PostOrderMemOutboard::create(data, IROH_BLOCK_SIZE).flip();
+            let hash = outboard.root;
+            (outboard.into_inner_with_prefix(), hash)
+        };
         outboard.splice(0..8, []);
```

Comment:
Is this still necessary? I tried to remove all length prefixes in bao-tree.

Reply:
you mean line 56? I don't actually know. My only hint on this change was replacing the function for the contents since there is no info in the crate/repo about why the helper functions are no longer necessary. Maybe I could look into that with some guidance there?

Reply:
ping @rklaehn